sos/plugins/openstack_tripleo.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2019 Red Hat, Inc., Cedric Jeanneret <cjeanner@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+
+
+class OpenStackTripleO(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """Log installation informations from OpenStack Installer
+    """
+
+    plugin_name = 'openstack_tripleo'
+
+    def setup(self):
+        # Notes: recursion is max 2 for container-puppet and tripleo-config
+        # Those directories are present on all OpenStack nodes and
+        # may contain sensitive data (password).
+        self.add_copy_spec([
+                '/var/log/paunch.log',
+                '/var/lib/container-puppet/',
+                '/var/lib/tripleo-config/'
+                ])
+
+    def postproc(self):
+        # Ensures we do not leak passwords from the tripleo-config location
+        # Usually, password are named as SERVICE_PASS or SERVICE_PASSWORD
+        # We do not have RSA keys in this location.
+        password_reg = r'("[a-zA-Z_]+PASS(WORD)?"):\s?"(.+)",?'
```

Comment:
can't we do better? At least to cover "secret" and "key"

Reply:
so those two apparently don't exist in the location, but I'll create a new method in "plugin" in order to actually mask secrets. It will be based on what we already have in oslo.utils and should be efficient.

Reply:
Presumably you're referring to the stuff in `oslo.utils.strutils`? Please discuss any changes or additions to the Plugin post-processing methods before going too far with this - if it's an extensive set of changes then that may justify a separate PR to focus on just that aspect - methods in `Plugin` need to be sufficiently generic to be useful for many plugins - otherwise if they are more tailored to the needs of one specific sub-family of plugins then we may want to consider a different organisation for the changes.

Reply:
@bmr-cymru indeed - but I've found an easier regexp that does match the current need, just have to clean the code and we're good.